### PR TITLE
Adding Visualizations

### LINKED ISSUE #14: t-sne visualization of embedding output

### PR DESCRIPTION
**Pull request by pieceOpiland**
Closes #14 

Adding various visualizations around training. Included are:

* Per epoch TSNE visualization over validation set per genders
* Logging of per ID silhouette scores
* Logging of per gender silhouette scores

It might be worth including per ID TSNE per epoch, but that can get noisy like the example I dropped in discord.

### CODE REVIEW

**Comment by pieceOpiland** [comment]
Note to self, improve the per speaker-id TSNE plots and upload a tuned model.

**Comment by richieyoum** [comment]
can you update the visualizations please :)

**Comment by pieceOpiland** [comment]
> can you update the visualizations please :)

Just updated them! I've included saved models from each epoch whose visualization is included.

I'm working on uploading an archive with all the artifacts to GA Tech's box. I'll include a shared link to it here when it's done uploading.

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ __pycache__/
 
 LibriSpeech/
 train-clean-100.tar.gz
+dev-clean.tar.gz
```
Comment (richieyoum) [inline .gitignore:7]: should we omit all .tar.gz files?

**Comment by richieyoum** [comment]
why is the silhouette score decreasing every epoch for gender?

File: tacotron2/layers.py [hunk 2]
```diff
@@ -48,7 +48,8 @@ def __init__(self, filter_length=1024, hop_length=256, win_length=1024,
         self.sampling_rate = sampling_rate
         self.stft_fn = STFT(filter_length, hop_length, win_length)
         mel_basis = librosa_mel_fn(
-            sampling_rate, filter_length, n_mel_channels, mel_fmin, mel_fmax)
+            sr=sampling_rate, n_fft=filter_length, n_mels=n_mel_channels, 
+            fmin=mel_fmin, fmax=mel_fmax)
```
Comment (richieyoum) [inline tacotron2/layers.py:52]: is there a reason for making this explicit in particular?

Comment (perrychu) [reply tacotron2/layers.py:52]: It was just throwing a future warning

**Comment by pieceOpiland** [comment]
> why is the silhouette score decreasing every epoch for gender?

My thoughts are that it's probably learning a feature space at a higher level than the original mel specs and that may be blurring some lines across genders?

**Comment by pieceOpiland** [comment]
> I'm working on uploading an archive with all the artifacts to GA Tech's box. I'll include a shared link to it here when it's done uploading.

It's out of date, but here's the link with the original artifacts: https://gatech.box.com/s/cx95iwjjm8pbhtn1dr8wl7hh271qr1i2

**Pull request merged**